pkg/service/connect_service.go

```diff
@@ -78,11 +78,13 @@ func (s *ConnectService) Serve(ctx context.Context, eval eval.IEvaluator) error
 		}
 		close(errChan)
 	}()
-	<-ctx.Done()
```

Comment:
Blocked here until flagd is shutdown (either gracefully or by an error elsewhere in the system), wholly unaware that Serve/ServeTLS may have returned an error.